### PR TITLE
Support engine events and detect config watcher failed

### LINKED ISSUE #1081: iNotify error when multiple Zilla instances are started in K8s Pods on a Portainer.io host
**Describe the bug**
Getting iNotify error when multiple Zilla instances are started in K8s Pods on a Portainer.io host. The issue is likely due to a limitation imposed by the host as described in this Portainer [forum post](https://forum.syncthing.net/t/omv6-portainer-docker-syncthing-inotify-problem/19539).

```
java.io.IOException: User limit of inotify instances reached or too many open files
	at java.base/sun.nio.fs.LinuxWatchService.<init>(LinuxWatchService.java:62)
	at java.base/sun.nio.fs.LinuxFileSystem.newWatchService(LinuxFileSystem.java:53)
	at io.aklivity.zilla.runtime.engine@0.9.82/io.aklivity.zilla.runtime.engine.internal.registry.FileWatcherTask.<init>(FileWatcherTask.java:52)
	at io.aklivity.zilla.runtime.engine@0.9.82/io.aklivity.zilla.runtime.engine.Engine.<init>(Engine.java:216)
	at io.aklivity.zilla.runtime.engine@0.9.82/io.aklivity.zilla.runtime.engine.EngineBuilder.build(EngineBuilder.java:147)
	at io.aklivity.zilla.runtime.command.start@0.9.82/io.aklivity.zilla.runtime.command.start.internal.airline.ZillaStartCommand.run(ZillaStartCommand.java:161)
	at io.aklivity.zilla.runtime.command@0.9.82/io.aklivity.zilla.runtime.command.internal.ZillaMain$Invoker.invoke(ZillaMain.java:69)
	at io.aklivity.zilla.runtime.command@0.9.82/io.aklivity.zilla.runtime.command.internal.ZillaMain.invoke(ZillaMain.java:40)
	at io.aklivity.zilla.runtime.command@0.9.82/io.aklivity.zilla.runtime.command.internal.ZillaMain.main(ZillaMain.java:34)
```

**Additional context**
The fix is likely to update the host setting, but in the majority of cases when a zilla image is deployed to a production cluster the running zilla instances won't need to watch the filesystem for changes. Any rollout of updates to a zilla config would result in tearing down pods and creating new ones with the new config. Adding a default option inside the container to not watch the filesystem for changes would also prevent this error from happening.

### PR DESCRIPTION
**Pull request by jfallows**
## Description

Creates a new `events` file for events at engine scope, in addition to per-core engine workers. Updates event reader to include these events as well so they are all picked up automatically by telemetry exporters.

Detect when `FileSystem.newWatchService()` fails, and record engine event, then still continue startup with observed `zilla.yaml` configuration.

Fixes #1081

### CODE REVIEW

**Review by akrambek — DISMISSED** [review]

File: runtime/engine/src/main/java/io/aklivity/zilla/runtime/engine/internal/event/EngineEventFormatter.java [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+import io.aklivity.zilla.runtime.engine.Configuration;
+import io.aklivity.zilla.runtime.engine.event.EventFormatterSpi;
+import io.aklivity.zilla.runtime.engine.internal.types.event.EngineConfigWatcherFailedExFW;
+import io.aklivity.zilla.runtime.engine.internal.types.event.EngineEventExFW;
+import io.aklivity.zilla.runtime.engine.internal.types.event.EventFW;
+
+public final class EngineEventFormatter implements EventFormatterSpi
+{
+    private static final String CONFIG_WATCHER_FAILED_FORMAT =
+        "Please restart to apply any config changes.";
```
Comment (vordimous) [inline runtime/engine/src/main/java/io/aklivity/zilla/runtime/engine/internal/event/EngineEventFormatter.java:29]:
```suggestion
        "The initial config loaded successfully, but the config watcher could not be created. Restart Zilla to apply any new config changes.";
```
It feels a bit wordy but will hopefully keep someone from getting into a restart loop.

**Review by vordimous — CHANGES_REQUESTED** [review]

**Review by vordimous — DISMISSED** [review]

File: runtime/engine/src/main/java/io/aklivity/zilla/runtime/engine/internal/event/EngineEventFormatter.java (hunk 1 above)
Comment (jfallows) [reply runtime/engine/src/main/java/io/aklivity/zilla/runtime/engine/internal/event/EngineEventFormatter.java:29]: Turns out that the config is not yet loaded successfully when this happens, leading to the above message followed by config validation failures for malformed `zilla.yaml` syntax.

Propose we go with this instead.
```suggestion
        "Dynamic config reloading is disabled.";
```
Then when an `IOException` occurs, we also have the exception message, such as:
```
ENGINE_CONFIG_WATCHER_FAILED Dynamic config reloading is disabled. User limit of inotify instances reached or too many open files.
```

**Review by vordimous — APPROVED** [review]
